pkg/controller/workload/kubernetes/resource/resource.go

```diff
@@ -365,8 +375,10 @@ func (c *secretClient) delete(ctx context.Context, template *corev1.Secret) erro
 		return nil
 	}
 
-	// We own the existing object. delete it.
-	return errors.Wrap(c.kube.Delete(ctx, remote), errDeleteSecret)
+	if err := c.kube.Delete(ctx, remote); err != nil {
+		return errors.Wrap(err, errDeleteResource)
+	}
+	return errors.New(errWaitingForDeletion)
```

Comment:
@muvaf this is likely inconsequential, but just wondering if we need to do this for secrets as well. It is unlikely that they hang or are tied to any external infrastructure. Seems like we could avoid some additional errors if we just were okay with making a delete call and not doing the check that they no longer exist. Open to leaving this how you have it if you feel strongly though. Might be easier to reason about later if we do the same thing for secrets and other resources